src/components/structures/RoomView.js

```diff
@@ -425,44 +389,9 @@ module.exports = React.createClass({
     },
 
     onMessageListScroll: function(ev) {
-        var sn = this._getScrollNode();
-        if (DEBUG_SCROLL) console.log("Scroll event: offset now:", sn.scrollTop, "recentEventScroll:", this.recentEventScroll);
-
-        // Sometimes we see attempts to write to scrollTop essentially being
-        // ignored. (Or rather, it is successfully written, but on the next
-        // scroll event, it's been reset again).
-        //
-        // This was observed on Chrome 47, when scrolling using the trackpad in OS
-        // X Yosemite.  Can't reproduce on El Capitan. Our theory is that this is
-        // due to Chrome not being able to cope with the scroll offset being reset
-        // while a two-finger drag is in progress.
-        //
-        // By way of a workaround, we detect this situation and just keep
-        // resetting scrollTop until we see the scroll node have the right
-        // value.
-        if (this.recentEventScroll !== undefined) {
-            if(sn.scrollTop < this.recentEventScroll-200) {
-                console.log("Working around vector-im/vector-web#528");
-                this._restoreSavedScrollState();
-                return;
-            }
-            this.recentEventScroll = undefined;
-        }
-
-        if (this.refs.messagePanel) {
-            if (this.state.searchResults) {
-                this.savedSearchScrollState = this._calculateScrollState();
-                if (DEBUG_SCROLL) console.log("Saved search scroll state", this.savedSearchScrollState);
-            } else {
-                this.savedScrollState = this._calculateScrollState();
-                if (DEBUG_SCROLL) console.log("Saved scroll state", this.savedScrollState);
-                if (this.savedScrollState.atBottom && this.state.numUnreadMessages != 0) {
-                    this.setState({numUnreadMessages: 0});
-                }
-            }
-        }
-        if (!this.state.paginating && !this.state.searchInProgress) {
-            this.fillSpace();
+        if (this.state.numUnreadMessages != 0 &&
+            this.refs.messagePanel.isAtBottom()) {
```

Comment:
Nit: Double-indent this please to avoid this line looking like it is the start of the block.

```javascript
if (this.state.numUnreadMessages != 0 &&
        this.refs.messagePanel.isAtBottom()) {
    this.setState({numUnreadMessages: 0});
```